### PR TITLE
update nix to 0.24.1

### PR DESCRIPTION
**Pull request by i509VCB**
This also enables minimal nix features

### CODE REVIEW

**Comment by Slabity** [comment]
Looks good to me, does this bump up the required rust version?

**Comment by i509VCB** [comment]
Will need to look if nix bumped msrv. The errors appear to be new clippy lints

**Comment by Slabity** [comment]
> Will need to look if nix bumped msrv. The errors appear to be new clippy lints

Thanks. Personally I don't really care if we bump the minimum version to whatever, but not sure if anyone using this project relies on an old compiler.

**Comment by i509VCB** [comment]
nix is apparently 1.46

**Comment by Drakulix** [comment]
> > Will need to look if nix bumped msrv. The errors appear to be new clippy lints
> 
> Thanks. Personally I don't really care if we bump the minimum version to whatever, but not sure if anyone using this project relies on an old compiler.

I think drm-rs is used in some embedded projects, which frequently don't have the newest rust compiler. But 1.46 *should* be reasonable, I think?

File: Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,11 @@ exclude = [".gitignore", ".github"]
 bitflags = "1"
 drm-ffi = { path = "drm-ffi", version = "0.2.1" }
 drm-fourcc = "^2.2.0"
-nix = "0.23.1"
+
+[dependencies.nix]
+version = "^0.24.1"
```
Comment (MarijnS95) [inline Cargo.toml]: Caret-matching is the default, fwiw.

File: src/control/mod.rs [hunk 2]
```diff
@@ -562,7 +562,10 @@ pub trait Device: super::Device {
 
         let prop_val_set = PropertyValueSet {
             prop_ids: unsafe { mem::transmute(prop_ids) },
-            prop_vals: unsafe { mem::transmute(prop_vals) },
+            prop_vals: unsafe {
+                #[allow(clippy::useless_transmute)] // TODO: Resolve clippy warning
+                mem::transmute(prop_vals)
```
Comment (MarijnS95) [inline src/control/mod.rs]: Should be fixed in `main` now: https://github.com/Smithay/drm-rs/pull/118/files#diff-cbe991c0ba9a31e65dfbb22e683b3a981f61154335714b511f54c8cd8d95ff12L565-R571

**Comment by Slabity** [comment]
Need to resolve merge conflicts and update `drm-ffi` as well to keep the two compatible.

**Comment by i509VCB** [comment]
> Need to resolve merge conflicts and update `drm-ffi` as well to keep the two compatible.

I have bumped nix in drm-ffi

**Pull request merged**